### ISSUE TITLE
Module was compiled with an incompatible version of Kotlin.

### ISSUE THREAD

**Issue by khs3994** [patrykandpatrick/vico#406]
**Describe the bug**
```kotlin
    // For Jetpack Compose.
    implementation("com.patrykandpatrick.vico:compose:1.12.0")

    // For `compose`. Creates a `ChartStyle` based on an M2 Material Theme.
    implementation("com.patrykandpatrick.vico:compose-m2:1.12.0")

    // For `compose`. Creates a `ChartStyle` based on an M3 Material Theme.
    implementation("com.patrykandpatrick.vico:compose-m3:1.12.0")

    // Houses the core logic for charts and other elements. Included in all other modules.
    implementation("com.patrykandpatrick.vico:core:1.12.0")
```
When I add a library and run, the following error occurs.
"Module was compiled with an incompatible version of Kotlin."
my kotlin version: 1.7.0
my compose version: 1.2.0

**Module**

- [x] `compose`
- [ ] `views`
- [x] `compose-m2`
- [x] `compose-m3`
- [x] `core` (select if none of the above apply)

**To reproduce**
add library and run

**Expected behavior**
No errors are expected

**Additional context**
full error msg
```
/Users/hyun_seung/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.9.10/dafaf2c27f27c09220cee312df10917d9a5d97ce/kotlin-stdlib-common-1.9.10.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.10/72812e8a368917ab5c0a5081b56915ffdfec93b7/kotlin-stdlib-1.9.10.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.10/72812e8a368917ab5c0a5081b56915ffdfec93b7/kotlin-stdlib-1.9.10.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.10/72812e8a368917ab5c0a5081b56915ffdfec93b7/kotlin-stdlib-1.9.10.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/transforms-3/4482a1155cb30be9e54fb057fbf7563f/transformed/core-1.12.0-api.jar!/META-INF/core_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/transforms-3/589f159f455eff9934cddd7a9d0b11a4/transformed/compose-m2-1.12.0-api.jar!/META-INF/compose-m2_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/transforms-3/5db4fd770e5bb3ef6ffdfca5a68971c6/transformed/compose-m3-1.12.0-api.jar!/META-INF/compose-m3_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

/Users/hyun_seung/.gradle/caches/transforms-3/b7ac72245e18252bbfb9106d311e14b8/transformed/compose-1.12.0-api.jar!/META-INF/compose_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.
```
pls help me...

**Comment by khs3994** [comment]
Please tell me the vico library version that corresponds to kotlin 1.7.0 and Compoe 1.2.0

> Hello! Vico 1.12.0 requires Kotlin 1.9.10 and version 2023.09.00 of the Jetpack Compose BOM.

**Comment by patrickmichalik** [comment]
That’d be Vico 1.4.2. Please note that this is an early version of the library, so it lacks many—if not most—of the features available in Vico 1.12.0. Additionally, dozens of bug fixes have been made since version 1.4.2. Kotlin 1.7.0 and Jetpack Compose 1.2.0 are over a year old, so I’d encourage you to update these dependencies and use a more recent version of Vico. If you opt to use Vico 1.4.2, please keep in mind that a lot of the information in the wiki and API reference won’t apply, and we will, unfortunately, be unable to accept bug reports from you.

**Comment by patrickmichalik** [comment]
I’ll be closing the issue. If you have any further questions, please let me know.
